catalogs/catalog-jdbc-doris/src/main/java/com/datastrato/gravitino/catalog/doris/operation/DorisTablePartitionOperations.java

```diff
@@ -0,0 +1,341 @@
+      } else {
+        throw new IllegalArgumentException("Unsupported partition type of Doris");
+      }
+
+      try (Statement statement = connection.createStatement()) {
+        statement.executeUpdate(
+            String.format(addPartitionSqlFormat, tableName, partition.name(), partitionValues));
+        return added;
+      }
+    } catch (SQLException e) {
+      throw exceptionConverter.toGravitinoException(e);
+    }
+  }
+
+  @Override
+  public boolean dropPartition(String partitionName) {
+    try (Connection connection = getConnection(databaseName)) {
+      String dropPartitionsSql =
+          String.format("ALTER TABLE `%s` DROP PARTITION `%s`", tableName, partitionName);
+      try (Statement statement = connection.createStatement()) {
+        statement.executeUpdate(dropPartitionsSql);
+        return true;
+      }
+    } catch (SQLException e) {
+      GravitinoRuntimeException exception = exceptionConverter.toGravitinoException(e);
+      if (exception instanceof NoSuchPartitionException) {
+        return false;
+      }
+      throw exception;
+    }
+  }
+
+  private Transform getPartitionInfo(Connection connection) throws SQLException {
+    String showCreateTableSql = String.format("SHOW CREATE TABLE `%s`", tableName);
+    try (Statement statement = connection.createStatement();
+        ResultSet result = statement.executeQuery(showCreateTableSql)) {
+      StringBuilder createTableSql = new StringBuilder();
+      while (result.next()) {
+        createTableSql.append(result.getString("Create Table"));
+      }
+      Optional<Transform> transform =
+          DorisUtils.extractPartitionInfoFromSql(createTableSql.toString());
+      return transform.orElseThrow(
+          () ->
+              new UnsupportedOperationException(
+                  String.format("%s is not a partitioned table", tableName)));
+    }
+  }
+
+  private Partition fromDorisPartition(
+      ResultSet resultSet, Transform partitionInfo, Map<String, Type> columnTypes)
+      throws SQLException {
+    String partitionName = resultSet.getString("PartitionName");
+    String partitionKey = resultSet.getString("PartitionKey");
+    String partitionValues = resultSet.getString("Range");
+    ImmutableMap.Builder<String, String> propertiesBuilder = ImmutableMap.builder();
+    propertiesBuilder.put("PartitionId", resultSet.getString("PartitionId"));
+    propertiesBuilder.put("VisibleVersion", resultSet.getString("VisibleVersion"));
+    propertiesBuilder.put("VisibleVersionTime", resultSet.getString("VisibleVersionTime"));
+    propertiesBuilder.put("State", resultSet.getString("State"));
+    propertiesBuilder.put("PartitionKey", partitionKey);
+    propertiesBuilder.put("DataSize", resultSet.getString("DataSize"));
+    propertiesBuilder.put("IsInMemory", resultSet.getString("IsInMemory"));
+    ImmutableMap<String, String> properties = propertiesBuilder.build();
```

Comment:
we should use a class such as `DorisPartitionPropertiesMetadata` to manage those reserved properties